pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -236,14 +225,108 @@ func initFSS(ctx context.Context, k8sClient clientset.Interface, controllerClust
 		}
 	}
 
-	// Set up kubernetes resource listeners for k8s orchestrator
+	var doesSvFssCRExist bool
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorGuest {
+		var isFSSCREnabled bool
+		// Check if csi-sv-feature-states-replication FSS exists and is enabled
+		if val, ok := k8sOrchestratorInstance.internalFSS.featureStates[common.CSISVFeatureStateReplication]; ok {
+			isFSSCREnabled, err = strconv.ParseBool(val)
+			if err != nil {
+				log.Errorf("Unable to convert %q to bool. csi-sv-feature-states-replication FSS disabled. Error: %v", val, err)
+				return err
+			}
+		} else {
+			log.Warn("csi-sv-feature-states-replication FSS not present")
+		}
+
+		// Initialize supervisor FSS map values in GC through CR if csi-sv-feature-states-replication FSS is enabled
+		if isFSSCREnabled {
+			cfg, err := common.GetConfig(ctx)
+			if err != nil {
+				log.Errorf("failed to read config. Error: %+v", err)
+				return err
+			}
+			// Get rest client config for supervisor
+			restClientConfig := k8s.GetRestClientConfigForSupervisor(ctx, cfg.GC.Endpoint, cfg.GC.Port)
+			// Get CNS operator client
+			cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restClientConfig, cnsoperatorv1alpha1.GroupName)
+			if err != nil {
+				log.Errorf("Failed to create CnsOperator client. Err: %+v", err)
+				return err
+			}
+
+			// Check if CnsCsiSvFeatureStates CR exists
+			svNamespace, err := cnsconfig.GetSupervisorNamespace(ctx)
+			if err != nil {
+				log.Errorf("Failed to retrieve supervisor cluster namespace from config. Err: %+v", err)
```

Comment:
start with lower case for error message? I think this applies in other places too

Reply:
Changed it